tests/store/test_runs_artifact_repo.py

```diff
@@ -0,0 +1,47 @@
+import pytest
+
+import mlflow
+from mlflow.exceptions import MlflowException
+from mlflow.store.runs_artifact_repo import RunsArtifactRepository
+from mlflow.store.s3_artifact_repo import S3ArtifactRepository
+
+from tests.projects.utils import tracking_uri_mock  # pylint: disable=unused-import
+
+
+@pytest.mark.parametrize("uri, expected_run_id, expected_artifact_path", [
+    ('runs:/1234abcdf1394asdfwer33/path/to/model', '1234abcdf1394asdfwer33', 'path/to/model'),
+    ('runs:/1234abcdf1394asdfwer33/path/to/model/', '1234abcdf1394asdfwer33', 'path/to/model/'),
+    ('runs:/1234abcdf1394asdfwer33', '1234abcdf1394asdfwer33', None),
+    ('runs:/1234abcdf1394asdfwer33/', '1234abcdf1394asdfwer33', None),
+    ('runs:///1234abcdf1394asdfwer33/', '1234abcdf1394asdfwer33', None),
+])
+def test_parse_runs_uri_valid_input(uri, expected_run_id, expected_artifact_path):
+    (run_id, artifact_path) = RunsArtifactRepository.parse_runs_uri(uri)
+    assert run_id == expected_run_id
+    assert artifact_path == expected_artifact_path
+
+
+@pytest.mark.parametrize("uri", [
+    'notruns:/1234abcdf1394asdfwer33/',  # wrong scheme
+    'runs:/',                            # no run id
+    'runs:1234abcdf1394asdfwer33/',      # missing slash
```

Comment:
URI scheme suggests that `/` is optional. No? Could this mean that `1234abcdf1394asdfwer33/` is a folder in the current directory and `run_uuid` is missing.

Reply:
discussed offline - we will reject the no slash case (i.e. leave this code as is) since the URI scheme definition neither mandates nor rejects that version and it is easier to add support for it later than to add it now and take it away.